CONTRIBUTING.md

```diff
@@ -170,6 +170,16 @@ for coding advice). The code must adhere to the following robustness principles
 are important for software that runs autonomously and continuously without direct
 interaction with a human (such as this Collector).
 
+### Method naming
+
+To keep naming patterns consistent across the project, naming patterns are enforced to make intent clear by:
+
+- Any method that allocates memory onto the heap must use the name prefix `New`
```

Comment:
What is the recommendation if it does not allocate memory on the heap?

Reply:
Hard to say in the general sense since methods that either pure or mutate state are bound by business logic.

I believe the best I could say that the method naming must reflect the work being done, keen to hear if you have any other ideas that pop out at you @bogdandrutu ?